Terminal.Gui/View/Layout/Pos.cs

```diff
@@ -0,0 +1,694 @@
+///                 </term>
+///                 <description>Creates a <see cref="Pos"/> object that can be used to center the <see cref="View"/>.</description>
+///             </item>
+///             <item>
+///                 <term>
+///                     <see cref="Absolute"/>
+///                 </term>
+///                 <description>
+///                     Creates a <see cref="Pos"/> object that is an absolute position based on the specified
+///                     integer value.
+///                 </description>
+///             </item>
+///             <item>
+///                 <term>
+///                     <see cref="Pos.Left"/>
+///                 </term>
+///                 <description>
+///                     Creates a <see cref="Pos"/> object that tracks the Left (X) position of the specified
+///                     <see cref="View"/>.
+///                 </description>
+///             </item>
+///             <item>
+///                 <term>
+///                     <see cref="Pos.X(View)"/>
+///                 </term>
+///                 <description>
+///                     Creates a <see cref="Pos"/> object that tracks the Left (X) position of the specified
+///                     <see cref="View"/>.
+///                 </description>
+///             </item>
+///             <item>
+///                 <term>
+///                     <see cref="Pos.Top(View)"/>
+///                 </term>
+///                 <description>
+///                     Creates a <see cref="Pos"/> object that tracks the Top (Y) position of the specified
+///                     <see cref="View"/>.
+///                 </description>
+///             </item>
+///             <item>
+///                 <term>
+///                     <see cref="Pos.Y(View)"/>
+///                 </term>
+///                 <description>
+///                     Creates a <see cref="Pos"/> object that tracks the Top (Y) position of the specified
+///                     <see cref="View"/>.
+///                 </description>
+///             </item>
+///             <item>
+///                 <term>
+///                     <see cref="Pos.Right(View)"/>
+///                 </term>
+///                 <description>
+///                     Creates a <see cref="Pos"/> object that tracks the Right (X+Width) coordinate of the
+///                     specified <see cref="View"/>.
+///                 </description>
+///             </item>
+///             <item>
+///                 <term>
+///                     <see cref="Pos.Bottom(View)"/>
+///                 </term>
+///                 <description>
+///                     Creates a <see cref="Pos"/> object that tracks the Bottom (Y+Height) coordinate of the
+///                     specified <see cref="View"/>
+///                 </description>
+///             </item>
+///         </list>
+///     </para>
+/// </remarks>
+public class Pos
```

Comment:
Making these records would reduce code for these, and it also makes the primary constructors make more sense.

Annoyingly, a primary constructor on a `class` has very different meaning and behavior from the primary constructor on a `record [class]`

Value equality, deconstruction to treat it like a tuple, and other goodies come along with a `record` for free. Roslyn source generators are behind that language feature.

# BUT!

Don't do that either, yet, because there are other fundamental changes I have suggestions for that would make that work pointless.